Pixel.py

```diff
@@ -1,119 +1,314 @@
-            col = (color.getRed(), color.getGreen(), color.getBlue())
-            self.image.putpixel((self.x, self.y), col)
+        """Set the color of a pixel
+
+        Parameters
+        ----------
+        color : Color
+            color to assign to pixel
+        """
+        self.image.putpixel((self.x, self.y), color.getRGB())
+
+    def setColorFrom(self, otherPixel):
+        """Set color of this pixel using color value from otherPixel
+
+        Parameters
+        ----------
+        otherPixel : Pixel
+            pixel to get color values from
+        """
+        self.setColor(otherPixel.getColor())
+
+    def updatePicture(self, alpha, red, green, blue):
+        """Update the picture based on the passed color values for this pixel
+
+        Parameters
+        ----------
+        alpha : int
+            transparancy value (CURRENTLY IGNORED)
+        red : int
+            red color value
+        green : int
+            green color value
+        blue : int
+            blue color value
+        """
+        #self.setAlpha(alpha)
+        self.setRed(red)
+        self.setGreen(green)
+        self.setBlue(blue)
 
-    def correctLevel(self, level):
-        return level%256
-
-# Class that for storing and doing cmputations with colors and RGB values.
-# Has getters and setters for the red, green, and blue portions of the over
-# all color as well as giving the user capabilities to compare color values.
-# Currently makeLighter and makeDarker do not work properly
 class Color:
+    """Class for storing and doing computations with colors and RGB values
+
+    Has getters and setters for the red, green, and blue portions of
+    the over all color as well as giving the user capabilities to
+    compare color values.  Currently makeLighter and makeDarker do not
+    work properly
+    """
+
+    wrapColorLevels = False
 
     def __init__(self, r, g=None, b=None):
+        """Initalize a color object
+
+        Parameters
+        ----------
+        r : list, tuple, Color or int
+            If r is a list or a tuple then r contains the RGB values,
+            if r is a Color object then use its color list for RGB values,
+            if r is an int then use it for all colors (graylevel)
+        g : int
+            green level (if not provided then r is used)
+        b : int
+            blue level (if not provided then r is used)
+        """
+        self.wrapColorLevels = JESConfig.CONFIG_WRAPPIXELVALUES != '0'
+
         if b == None:
-            if isinstance(r, Color):
+            if isinstance(r, tuple):
+                self.color = list(r)
+            elif isinstance(r, list):
                 self.color = r
+            elif isinstance(r, Color):
+                self.color = r.color
             else:
-                val = Pixel.correctLevel(r)
-                self.color = (val, val, val)
+                val = self.correctLevel(r)
+                self.color = [val, val, val]
         else:
```

Comment:
Do we need to do checking if we pass in both a red and green value, but no blue value? May cause issue to use correctLevel on a variable that is equal to None